drivers/amplifiers/ada4250/ada4250.c

```diff
@@ -164,15 +165,22 @@ int32_t ada4250_soft_reset(struct ada4250_dev *dev)
  */
 int32_t ada4250_en_refbuf(struct ada4250_dev *dev, bool refbuf)
 {
+	int32_t ret;
+
 	switch(dev->device_id) {
 	case ADA4250:
-		return ada4250_update(dev, ADA4250_REG_REFBUF_EN, ADA4250_REFBUF_MSK,
-				      ADA4250_REFBUF(refbuf));
+		ret = ada4250_update(dev, ADA4250_REG_REFBUF_EN, ADA4250_REFBUF_MSK,
+				     ADA4250_REFBUF(refbuf));
 	case ADA4230:
-		return gpio_set_value(dev->gpio_bufen, dev->refbuf_en);
+		ret = gpio_set_value(dev->gpio_bufen, dev->refbuf_en);
 	default:
 		return -ENODEV;
 	}
+
+	if (ret == SUCCESS)
```

Comment:
The proposed change is good and mergeable but perhaps you could change a bit the phrasing of this code. I noticed the team currently uses this approach:
https://github.com/analogdevicesinc/no-OS/blob/master/drivers/frequency/hmc7044/hmc7044.c#L288-L294

So if there is an error, quit.
Then set any variable for the no-error scenario.
Finally return SUCCESS or return 0.

It's obviously not mandatory, it's more style related. From my side I'm approving as-is but if you wanna get the style homogenous with rest of noOS code it's up to you.

Reply:
Thanks for the information. I'll update the code with the current styling.